File: _CodeTopics/LeetCode_contest/weekly/weekly2021/262/262_1.py
class Solution(object):
    def twoOutOfThree(self, nums1, nums2, nums3):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :type nums3: List[int]
        :rtype: List[int]
        """
        
        dic, res = {}, set()
        for arr in [nums1, nums2, nums3]:
            arr = set(arr)
            for elem in arr:
                dic[elem] = dic.setdefault(elem, 0) + 1
                if dic[elem] >= 2:
                    res.add(elem)
        return list(res)
    
"""
https://leetcode-cn.com/submissions/detail/227322774/

288 / 288 个通过测试用例
状态：通过
执行用时: 36 ms
内存消耗: 12.9 MB
"""

File: _CodeTopics/LeetCode/2001-2200/002032--weekly262_1/tran/262_1.cpp
class Solution {
public:
    vector<int> twoOutOfThree(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3) {

        // 第 262 场周赛第一题
        // https://leetcode.cn/submissions/detail/227322774/

        unordered_map<int, int> dic;
        unordered_set<int> res;
        for (auto& arr_vec : {nums1, nums2, nums3}) {
            unordered_set<int> arr(arr_vec.begin(), arr_vec.end());
            for (auto elem : arr) {
                if (dic.find(elem) != dic.end()) {
                    dic[elem]++;
                } else {
                    dic[elem] = 1;
                }
                if (dic[elem] >= 2) {
                    res.insert(elem);
                }
            }
        }
        return vector<int>(res.begin(), res.end());
    }
};

/*
https://leetcode.cn/submissions/detail/391623757/

执行用时：
28 ms
, 在所有 C++ 提交中击败了
11.57%
的用户
内存消耗：
27.8 MB
, 在所有 C++ 提交中击败了
8.79%
的用户
通过测试用例：
288 / 288
*/
